code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingCacheUtilities.java

```diff
@@ -0,0 +1,304 @@
+/*
+  Copyright 2021 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile;
+
+import static com.adobe.marketing.mobile.MessagingConstants.CACHE_NAME;
+import static com.adobe.marketing.mobile.MessagingConstants.IMAGES_CACHE_SUBDIRECTORY;
+import static com.adobe.marketing.mobile.MessagingConstants.LOG_TAG;
+import static com.adobe.marketing.mobile.MessagingConstants.MESSAGES_CACHE_SUBDIRECTORY;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class contains functionality to cache the json message payload and any image asset URL's present in an
+ * AJO in-app message.
+ */
+class MessagingCacheUtilities {
+    private final static String SELF_TAG = "MessagingCacheUtilities";
+    private final int STREAM_WRITE_BUFFER_SIZE = 4096;
+    private final SystemInfoService systemInfoService;
+    private final NetworkService networkService;
+    private final Map<String, String> assetMap = new HashMap<>();
+    private CacheManager cacheManager;
+
+    MessagingCacheUtilities(final SystemInfoService systemInfoService, final NetworkService networkService) throws MissingPlatformServicesException {
+        this.systemInfoService = systemInfoService;
+        this.networkService = networkService;
+        if (networkService == null) {
+            throw new MissingPlatformServicesException("Network service implementation missing");
+        }
+        cacheManager = new CacheManager(systemInfoService);
```

Comment:
Instead of instantiating Cachemanager in the constructor, let's pass it as a parameter to constructor. It will help in mocking in unit testing as well.